### PR TITLE
ROS#209: Demo application

### PR DESCRIPTION
**Pull request by SteelPh0enix**
This PR contains remaining SAMV71 demo application implementation and Python script that allows to interface with the SAMV71 app.

### CODE REVIEW

File: demos/samv71-spi-accelerometer/src/main.rs [hunk 1]
```diff
@@ -119,64 +166,196 @@ static EVENT_START_STORAGE: EventStorage = EventStorage::new();
...
+
+fn init_imu(spi: Spi<SPI0, Master>) -> IMU {
+    let mut imu = match IMU::new(spi) {
+        Ok(imu) => imu,
+        Err(reason) => panic!("Could not initialize IMU: {reason:?}"),
+    };
+
+    logln!(
+        "Is IMU responsive: {}",
+        if imu.is_alive().unwrap() { "yes" } else { "no" }
```
[the remark this answers is not in the document]
Comment (SteelPh0enix) [reply demos/samv71-spi-accelerometer/src/main.rs:332]: True, there's no point in that, changed this to `panic!` on error

File: demos/samv71-spi-accelerometer/src/main.rs [hunk 2]
```diff
@@ -45,19 +66,46 @@ use aerugo::{
...
 };
+use lsm6dso::config::control::AccelerometerTestMode;
+use lsm6dso::config::control::GyroscopeTestMode;
+use lsm6dso::{
+    config::fifo::config::{
+        AccelerometerBatchingRate, DataRateChangeBatching, FifoConfig, FifoMode,
+        FifoWatermarkThreshold, GyroscopeBatchingRate, StopOnWatermarkThreshold,
+    },
+    LSM6DSO,
+};
```
[the remark this answers is not in the document]
Comment (SteelPh0enix) [reply demos/samv71-spi-accelerometer/src/main.rs:84]: done

File: demos/samv71-spi-accelerometer/src/main.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (SteelPh0enix) [reply demos/samv71-spi-accelerometer/src/main.rs:332]: not really, changed it to panic!

File: src/tasklet.rs [hunk 3]
```diff
@@ -17,7 +17,7 @@ mod tasklet_status;
 mod tasklet_storage;
 mod tasklet_vtable;
 
-pub(crate) use self::tasklet_id::TaskletId;
+pub use self::tasklet_id::TaskletId;
```
[the remark this answers is not in the document]
Comment (SteelPh0enix) [reply src/tasklet.rs:20]: moved

File: demos/samv71-spi-accelerometer/src/main.rs [hunk 4]
```diff
@@ -119,64 +166,196 @@ static EVENT_START_STORAGE: EventStorage = EventStorage::new();
...
+    StopMeasurements,
+    TransmitImuData,
+    UartReader,
+}
+
+pub type TaskletMap = [(DemoTaskletName, TaskletId); 8];
+
+/// This is a mapping of app's tasklet IDs to Aerugo's tasklet IDs.
+/// Should be filled by the system @ init.
+pub static mut TASKLET_MAP: Option<TaskletMap> = None;
```
[the remark this answers is not in the document]
Comment (SteelPh0enix) [reply demos/samv71-spi-accelerometer/src/main.rs]: moved to tasklet context

**Review by Glamhoth — APPROVED** [review]

**Pull request merged**